content/blaze.md

```diff
@@ -0,0 +1,560 @@
+  <div>{{name}} gets a {{color}} card!</div>
+{{/let}}
+```
+
+Note that `name` and `color` (and `todo` above) are only added to scope in the template, they *are not* added to the data context. Specifically this means if you call a helper, they will not be on `this`. So if you need to access them in a helper, you should pass them in as an argument (like we do with `(todoArgs todo)` above).
+
+<h4 id="each-and-with">Each and With</h4>
+There are also two Spacebars builtin that are very common for historical reasons but we discourage the use of (see [use each-in](#use-each-in) below) as they change the data context *within* a template, which is difficult to reason about.
+
+Like `{{#each .. in}}`, `{{#each}}` iterates over an array or cursor, changing the data context within its content block to be the item iterated over. `{{#with}}` simply changes the data context inside itself. In most cases it's better to use `{{#each .. in}}` and `{{#let}}` to achieve the purposes that these are commonly used for.
+
+<h4 id="strictness">Strictness</h4>
+Spacebars starts from a very strict interpretation of HTML. In particular you need to careful about self-closing tags and when you are allowed to use them. For instance, you can't self-close a `div` (`<div/>`) in Spacebars, and you need to close some tags that a browser might not require you to (such as a `<p>` tag).
+
+<h4 id="escaping">Escaping</h4>
+To insert a literal {{, {{{, or any number of curly braces, put a vertical bar after it. So `{{|` will show up as `{{`, `{{{|` will show up as `{{{`, and so on.
+
+<h2 id="reusable-components">Creating reusable components in Blaze</h2>
+In <a href="ui-ux">the UI/UX article</a> we discussed the merits of creating reusable components that are modular and depend and interact with their environment in clear and minimal ways. 
+
+Although Blaze, as a simple template-based runtime, doesn't enforce a lot of these principles (as some other rendering frameworks, such as React, do) you can take steps to write your Blaze components in such a way that you can enjoy most of the same benefits. This section will attempt to outline some "best practice" for writing such Blaze components.
+
+Examples below will reference the `listShow` component from the Todos example app.
+
+<h3 id="validate-data-context">Validate data contexts</h3>
+In order to ensure your component is only used in the way you expect, you should validate the data context provided to it. The data context provides the inputs to the component and so should be checked.
+
+XXX: this does not yet work
+
+You can do this in the components `onCreated()` callback, like so:
+
+```js
+Template.listsShow.onCreated(function() {
+  this.autorun(() => {
+    check(Template.data(), new SimpleSchema({
+     list: {blackbox: true},
+     todosReady: {type: Boolean}
+   }));
+  });
+```
+
+By placing the validation in an `autorun()` we ensure that even if the data contexts reactively changes, it always fits the expected schema.
```

Comment:
We use an `autorun()` to make sure that the data context is validated again whenever it changes.